packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -274,36 +271,54 @@ class Auth {
     }
 
     /**
-     * WARNING: This function is relevant to be used in Hybrid deployments only.
-     * Compares the refresh_token keys for guest('cc-nx-g') and registered('cc-nx') login from the cookie received from SFRA with the copy stored in localstorage on PWA Kit
-     * to determine if the login state of the shopper on SFRA site has changed. If the keys are different we return true considering the login state did change. If the keys are same,
-     * we compare the values of the refresh_token to cover an edge case where the login state might have changed multiple times on SFRA and the eventual refresh_token key might be same
-     * as that on PWA Kit which would incorrectly show both keys to be the same even though the sessions are different.
-     * @returns {boolean} true if the keys do not match (login state changed), false otherwise.
+     * Returns the SLAS access token or an empty string if the access token
+     * is not found in local store or if SFRA wants PWA to trigger refresh token login.
+     *
+     * On PWA-only sites, this returns the access token from local storage.
+     * On Hybrid sites, this checks whether SFRA has sent an auth token via cookies.
+     * Returns an access token from SFRA if it exist.
+     * If not, the access token from local store is returned.
+     *
+     * This is only used within this Auth module since other modules consider the access
+     * token from this.get('access_token') to be the source of truth.
+     *
+     * @returns {string} access token
      */
-    private hasSFRAAuthStateChanged() {
-        const refreshTokenKey =
-            (this.get('refresh_token_registered') && 'refresh_token_registered') ||
-            'refresh_token_guest'
+    private getAccessToken() {
+        let accessToken = this.get('access_token')
+        const sfraAuthToken = this.get('access_token_sfra')
+
+        if (sfraAuthToken && accessToken !== sfraAuthToken) {
+            /*
+             * If SFRA sends 'refresh', we return an empty token here so PWA can trigger a login refresh
+             * This key is used when logout is triggered in SFRA but the redirect after logout
+             * sends the user to PWA.
+             */
+            if (sfraAuthToken === 'refresh') {
+                this.set('access_token', '')
+                this.clearSFRAAuthToken()
+                return ''
+            }
```

Comment:
Is there a scenario where `sfraAuthToken === 'refresh'` and `accessToken !== sfraAuthToken`? I'm thinking there might not be a scenario when this is happening, and if that is the case we can probably pull this code out of the wrapping condition to look more like this to help with readability. 

```
if (sfraAuthToken === 'refresh') {
    this.set('access_token', '')
    this.clearSFRAAuthToken()
    return ''
}

   ...
```

Reply:
`accessToken` will never be `refresh` so all scenarios where we enter the `sfraAuthToken === 'refresh'` block is also where `accessToken !== sfraAuthToken`. 

But you are correct that readability could be improved in this area. To do that, I can remove the `accessToken !== sfraAuthToken` condition since if `sfraAuthToken === accessToken`, there should be no changes to the customer id, usid or customer type since it's the same token. So going into this branch if the two match would not change anything.